batch-submitter/drivers/proposer/driver.go

```diff
@@ -219,12 +218,20 @@ func (d *Driver) GetBatchBlockRange(
 	}
 	start.Add(start, blockOffset)
 
-	backend, _ := rollup.NewBackend("l1")
-	index, err := d.cfg.RollClient.GetLatestTransactionIndex(backend)
+	currentHeader, err := d.cfg.L1Client.HeaderByNumber(ctx, nil)
 	if err != nil {
 		return nil, nil, err
 	}
-	end := new(big.Int).SetUint64(*index)
+	finality := new(big.Int).SetUint64(d.cfg.FinalityConfirmations)
+	finality.Add(finality, new(big.Int).SetInt64(2)) // add 2 block number buffer to dtl sync data
```

Comment:
This `2` buffer recommended to define it as a constant

Reply:
done